File: Entities/ActivityEntities/Activity.cs
using PMS.Entities.DepartmentEntities;

namespace PMS.Entities.ActivityEntities
{
    public class Activity
    {
       
        public int Id { get; set; }
        public int DepartmentId { get; set; }
        public Department? Department { get; set; }
        public string Subject { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int ActivityTypeId { get; set; }
        public ActivityType? Type { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public int CreatedById { get; set; }
        public AppUser? CreatedBy { get; set; }
        public ActivityLevel Level { get; set; } = ActivityLevel.Normal;
        public List<ActivityAssignee> Assignees { get; set; }
        public List<ActivityComment> Comments { get; set; }

        public Activity NewEskeyActivity(string location, string userId) {
            return new Activity
            {
                Subject = location + " Eskeys",
                DepartmentId = 1,
                Description = "Eskey Delivery from " + location + " created",
                ActivityTypeId = 4,
                CreatedAt = DateTime.Now,
                CreatedById = int.Parse(userId),
                Level = ActivityLevel.Important,
            };
        }

    }
}

File: ClientApp/src/app/models/activityAgg/activity.ts


import { ActivityLevel } from './activity.level';
import { ActivityType } from './activityType';


export interface Activity {
  id: number;
  subject: string;
  departmentId: number;
  departmentName: string;
  description: string;
  activityTypeId: number;
  type: ActivityType;
  createdAt: string;
  createdById: number;
  typeName: string;
  createdByName: string;
  createdByUsername: string;
  level: ActivityLevel;
  assignees: Assignee[];
  comments: ActivityComment[];
}


export interface Assignee {
  assigneeId: number;
  assigneeUserName: string;
  isSelected: boolean;
}

export interface ActivityComment {
  commentorId: number;
  activityId: number;
  commentorUserName: string;
  comment: string;
  createdAt?: string;
}
